src/components/BaseSearch.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="base-search">
+    <input
+      v-model="localSearchValue"
+      type="text"
+      :placeholder="placeholder"
+      class="base-search__input"
+    />
+    <img
+      src="@/assets/icons/search.svg"
+      alt="SearchIcon"
+      class="base-search__icon-search"
+    />
+    <img
+      v-show="localSearchValue !== ''"
+      src="@/assets/icons/cross.svg"
+      alt="ClearIcon"
+      class="base-search__icon-cancel"
+      @click="clearSearch"
+    />
```

Comment:
for accessibility reasons would be better to wrap an image into a button component, so that it can be properly read by screen readers.
(just general feedback, pls dont fix it)